### PR TITLE
Remove obsolete pthread_setconcurrency call

### PR DESCRIPTION
**Pull request by DaoWen**
The function is a no-op in pthreads implementations on modern systems,
and including the call requires extra configuration to avoid compiler
warnings when building on Linux.

### CODE REVIEW

**Comment by agrippa** [comment]
Nice. Looks good to me. @budimlic, you should take a look at this and merge it once Nick merges CONTRIBUTING.md in.

**Review by budimlic — APPROVED** [review]

**Pull request merged**
